src/service/auth.rs

```diff
@@ -64,7 +64,7 @@ impl AuthService {
         let mut request = AttributeContext_Request::default();
         let mut http = AttributeContext_HttpRequest::default();
         let headers: HashMap<String, String> = hostcalls::get_map(MapType::HttpRequestHeaders)
-            .unwrap()
+            .expect("failed to retrieve HttpRequestHeaders from host")
```

Comment:
Let's leave this for now (and the other 2 calls below, as well as the one in `rate_limit.rs`), but I think this should eventually become some kind of `FailureMode` dependent behavior...

Reply:
100% agreed again here

Reply:
The expect adds some info, but still causes a panic...which is what we're looking to avoid, no?

Reply:
maybe unwrap_or() and provide some sane default (an empty map?)?

Reply:
The primary problem is this should perform an action based on the `FailureMode` defined for the service - if this service is set to `Deny` then we don't want to populate the request with a default value and let it through. This is a much larger refactor though than can be solved in this PR

Reply:
>provide some sane default (an empty map?)?

there is no "sane default" in this case, as it will just propagate the error to the next inline... i.e. in this case probably the CEL expression is going to fail. This error here tho, would mean something went wrong with envoy. 

As @adam-cattermole rightly pointed out, this would need to fallback to some user configured behavior (analogous to e.g. Limitador being unreachable). So we need to get rid of this one when we figured out that part of the issue (see [this "issue"](https://github.com/Kuadrant/kuadrant-operator/issues/1031) for more context